src/flow-manager.c

```diff
@@ -248,6 +254,51 @@ static int FlowManagerFlowTimeout(const Flow *f, enum FlowState state, struct ti
 
     /* do the timeout check */
     if (flow_times_out_at >= ts->tv_sec) {
+        /* for bypassed flow we need to check if the capture method has seen
```

Comment:
Flow is not locked at this point. I think we should lock it inside the util func I suggested, which should only be called for bypassed flows.

Reply:
actually, it seems better to call it from ```FlowManagerFlowTimedOut``` instead, which is called after the flow is locked.